gemrb/GUIScripts/iwd/CharGen.py

```diff
@@ -535,49 +548,17 @@ def SetCharacterDescription():
 					TextArea.Append ("+")
 					j = j + 1
 				TextArea.Append ("\n")
-
-		if MageSpell !="*":
-			TextArea.Append ("\n")
-			TextArea.Append (11027)
-			TextArea.Append (":\n")
-			t = GemRB.GetPlayerStat (MyChar, IE_ALIGNMENT)
-			Learnable = Spellbook.GetLearnableMageSpells (GemRB.GetPlayerStat (MyChar, IE_KIT), t,1)
-			MageSpellBook = GemRB.GetVar ("MageSpellBook")
-			MageMemorized = GemRB.GetVar ("MageMemorized")
-			for i in range (len(Learnable)):
-				if (1 << i) & MageSpellBook:
-					Spell = GemRB.GetSpell (Learnable[i])
-					TextArea.Append (Spell["SpellName"])
-					if (1 << i) & MageMemorized:
-						TextArea.Append (" +")
-					TextArea.Append ("\n")
-
-		if PriestSpell == "*":
-			PriestSpell = DruidSpell
-		if PriestSpell!="*":
-			TextArea.Append ("\n")
-			TextArea.Append (11028)
-			TextArea.Append (":\n")
-			t = GemRB.GetPlayerStat (MyChar, IE_ALIGNMENT)
-			if PriestSpell == "MXSPLPRS" or PriestSpell == "MXSPLPAL":
-				ClassFlag = 0x4000
-			elif PriestSpell == "MXSPLDRU" or PriestSpell == "MXSPLRAN":
-				ClassFlag = 0x8000
-			else:
-				ClassFlag = 0
-
-			Learnable = Spellbook.GetLearnablePriestSpells( ClassFlag, t, 1)
-			PriestMemorized = GemRB.GetVar ("PriestMemorized")
-			for i in range (len(Learnable)):
-				if (1 << i) & PriestMemorized:
-					Spell = GemRB.GetSpell (Learnable[i])
-					TextArea.Append (Spell["SpellName"])
-					TextArea.Append (" +\n")
 	return
 
+def GetClassFlag(TableName):
+	if TableName in ("MXSPLPRS", "MXSPLPAL"):
+		return 0x4000
+	elif TableName in ("MXSPLDRU", "MXSPLRAN"):
+		return 0x8000
+	else:
```

Comment:
these are swapped now

Reply:
Huh, they are swapped… and they are not. It turns out there are three places where it `ClassFlag` was computed, and I didn't notice these snippets actually differ [here](https://github.com/gemrb/gemrb/blob/8d7e141aa0dedffd361ce4ead6fe5a5775d2fa4d/gemrb/GUIScripts/iwd/CharGen.py#L341), [here](https://github.com/gemrb/gemrb/blob/8d7e141aa0dedffd361ce4ead6fe5a5775d2fa4d/gemrb/GUIScripts/iwd/CharGen.py#L562), and [here](https://github.com/gemrb/gemrb/blob/8d7e141aa0dedffd361ce4ead6fe5a5775d2fa4d/gemrb/GUIScripts/iwd/CharGen.py#L1582) (I've extracted third one). It should be correct, though? `0x4000` is supposed to be mask for cleric spells as I see?

Reply:
Looking at the files, this version is fine, so you accidentally fixed something. :)